src/assemble_workflow/dists.py

```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+from assemble_workflow.dist import Dist, DistZip, DistTar, DistRpm
+
+class Dists:
+    DISTRIBUTIONS = {
+        "rpm": DistRpm,
+        "tar": DistTar,
+        "zip": DistZip,
+    }  
+
+    @classmethod
+    def create_dist(cls, name: str, path: str, min_path: str, distribution: str) -> Dist:
+        if distribution not in cls.DISTRIBUTIONS:
+            raise ValueError("Distribution not specified or invalid distribution")
```

Comment:
`raise ValueError("Invalid distribution: ${distribution}")` so we know what was passed in